### ISSUE TITLE
Add GaussianMixtureMask transform

### ISSUE THREAD

**Issue by guarin** [lightly-ai/lightly#1661]
See https://github.com/lightly-ai/lightly/issues/1646 for information

**Comment by eleensmathew** [comment]
Hi can I work on this? Also can you mention more details about the issue?

**Comment by guarin** [comment]
Yes of course :) Give me one sec to add some more info.

**Comment by guarin** [comment]
Hi @eleensmathew did you already have time to look into this?

**Comment by eleensmathew** [comment]
Sorry but i became busy. Feel free to unassign me from this issue.

**Comment by snehilchatterjee** [comment]
Hello this actually looks a little interesting. Am I correct in assuming that we have to implement this one ourself ?

**Comment by guarin** [comment]
Yes I could not find any other implementation. From the paper:
![Screenshot 2024-10-08 at 16 24 53](https://github.com/user-attachments/assets/97574c4b-0f0e-4bb5-8010-42a6e0002340)

**Comment by snehilchatterjee** [comment]
Then I'd like to work on this. But is it okay if I take my time with this because I am going to be having my exams soon ?

**Comment by guarin** [comment]
Of course, no problem

**Comment by snehilchatterjee** [comment]
Hey @guarin
This is the result I am getting for the following snippet. I think this is looks very similar to what they have in the paper. This is for c=20..As we decrease c the image get more and more blurry.
```python
def gaussian_kernel(size, sigma, center, device='cpu'):
    u, v = torch.meshgrid(torch.arange(0, size[0], device=device), torch.arange(0, size[1], device=device))
    u0, v0 = center
    gaussian = torch.exp(- ((u - u0) ** 2 / (2 * sigma[0] ** 2) + (v - v0) ** 2 / (2 * sigma[1] ** 2)))
    return gaussian
```
![image](https://github.com/user-attachments/assets/670cba52-1ac1-44bb-8af9-ffcf06f47ec3)

From the paper:
![image](https://github.com/user-attachments/assets/1f8bc180-3ab8-4012-beee-5357f449839f)

**Comment by JanumalaAkhilendra** [comment]
I would Like to work on this issue , if it is not assigned to anyone . 
Thank you :)

**Comment by guarin** [comment]
@snehilchatterjee this looks great! Do you have time to create a PR for it? And do you have examples of the image getting blurrier with increasing augmentation strength? Then we could check if it really is the same as in the paper.

@JanumalaAkhilendra let's check if @snehilchatterjee would like to create the PR as he has already been working on the issue. We also have many other good first issues: https://github.com/lightly-ai/lightly/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22

**Comment by snehilchatterjee** [comment]
Yes I will create a PR for it soon. And I will also send some example images shortly.

**Comment by snehilchatterjee** [comment]
Hi @Guarin, during further testing, I noticed that the mask generated by my approach doesn’t match the one produced by their method. Additionally, my results appear to be darker. I will investigate this issue further and get back to you soon.

**Comment by payo101** [comment]
@snehilchatterjee Hi, can you ping your discord id? I wanted some help in getting sample input-output pairs for my transform actually. Could you help with that?

**Comment by snehilchatterjee** [comment]
yeah sure my username is `snehilchatterjee`
